src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -554,16 +560,14 @@ private BoundLambda ReallyInferReturnType(NamedTypeSymbol delegateType, Immutabl
 
         public BoundLambda BindForReturnTypeInference(NamedTypeSymbol delegateType)
         {
-            var cacheKey = ReturnInferenceCacheKey.Create(delegateType, IsAsync);
-
-            BoundLambda result;
-            if (!_returnInferenceCache.TryGetValue(cacheKey, out result))
+            if (_returnInferenceCache.TryGetValue(delegateType, out var result))
```

Comment:
>var [](start = 68, length = 3)

Please spell out the type, or better revert the inlining of the declaration. In general, ```var``` is not welcome in compilers code base.